File: TG-V1/deck.cpp
#include "deck.h"
#include "algorithm"

using namespace std;

const char* VALUES[] = {"Ace", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten", "Queen", "Jack", "King"};
const char* SUITS[] = {"diamonds", "spades", "hearts", "clubs"};

void Card::Print()
{
    printf("%s of %s", VALUES[value], SUITS[suit]);
}

void Card::SetValues(int val, int su)
{
    value = val;
    suit = su;
    rng = rand();
}

bool compare_card_rng(Card a, Card b)
{
    return a.rng < b.rng;
    printf("%d is less than %d\n", a.rng, b.rng);
}

void Deck::Shuffle()
{
    sort(deck, deck + 52, compare_card_rng);
}

File: TG-V2/Util/Models/Deck.cs
using System;

namespace Util.Models
{
    public class Deck
    {
        private Random rng = new Random();
        public Card[] Cards { get; set; } // 4 decks of 52 cards
        public Deck(int numDecks = 4)
        {
            Cards = new Card[numDecks * 52];
            int index = 0;
            for (int card = 1; card <= 13; card++)
                for (int suit = 1; suit <= 4; suit++)
                    for (int i = 0; i < numDecks; i++)
                    {
                        Cards[index] = new Card(card, suit);
                        index++;
                    }
        }

        public void Shuffle() => Cards.Shuffle();
    }
}